resources/tag_models/event.json

```diff
@@ -0,0 +1,29 @@
+{
+  "tag": {
+	"api_key": "", // String - API key - Make sure the request is from a valid client (idea nicked from Keen.io) useful if a page gets copied onto a Russian website and creates noise
+	"version": "", // Version of the tracking client e.g. "JS v1.2"
+	"type": "event", // Type of request (page or event)
+	"id": "", // ID of this request, must be unique
+	"offset": 0 // Delay of this tag between event happening and being sent to server - milliseconds
+  },
+
+  "event": {
+	"page_id": "", // ID of the page on which this event happened.
+	"parent_id": "", // Parent event ID if this is a child event.
```

Comment:
Understanding
I don't have time to do a survey, our current supplier could be bust in a month.
The normal use-case is still simple, just ping an event and don't worry about ID's.
There are specific use-cases where the relationship is needed, which will involve a defined statement of work, and user story, and testing, and assistance, and the rest of it.

Implementation
At least by having a guid generating component, only the component has to remember the "parent" id for the next event it sends - it's not shared state.

Algorithms
It's got to be achievable in SQL, there are a team of 30-40 people who know sql, plus it has to work with an off-the-shelf reporting tool which knows sql and how to do joins.

Finally
We do have to maintain the status quo too, millions in revenue are attached to this data, it's used everywhere - going into reports and more systems.
It's beyond the scope of this project to change every report, every business process, every third party system to accept a new model.